### PR TITLE
Resolves symbolic links for CUDA paths

### LINKED ISSUE #35122: undeclared inclusion(s) in rule '@nccl_archive//:device_lib'
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux centos 6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source
- TensorFlow version: 2.0.0
- Python version: 
- Installed using virtualenv? pip? conda?:
- Bazel version (if compiling from source): 0.26.1
- GCC/Compiler version (if compiling from source): 5.4
- CUDA/cuDNN version: 10.0.130/7.6.4
- GPU model and memory:

**Describe the problem**

```
ERROR: /home/conda/.cache/bazel/_bazel_conda/6f8a02c56728b792c78882b32cc69e39/external/nccl_archive/BUILD.bazel:53:1: undeclared inclusion(s) in rule '@nccl_archive//:device_lib':
this rule is missing dependency declarations for the following files included by 'external/nccl_archive/src/collectives/device/max_f64_reduce.cu.cc':
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/cuda_runtime.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/crt/host_config.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/builtin_types.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/device_types.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/crt/host_defines.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/driver_types.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/vector_types.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/surface_types.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/texture_types.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/library_types.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/channel_descriptor.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/cuda_runtime_api.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/cuda_device_runtime_api.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/driver_functions.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/vector_functions.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/vector_functions.hpp'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/crt/common_functions.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/crt/math_functions.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/crt/math_functions.hpp'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/cuda_surface_types.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/cuda_texture_types.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/crt/device_functions.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/crt/device_functions.hpp'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/device_atomic_functions.h'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/device_atomic_functions.hpp'
  '/usr/local/cuda-10.0/targets/x86_64-linux/include/crt/device_double_functions.h'

```

**Provide the exact sequence of commands / steps that you executed before running into the problem**
```bash
bazel build //tensorflow:libtensorflow_cc.so
```

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

### PR DESCRIPTION
**Pull request by njzjz**
Resolves #35122.

### CODE REVIEW

**Comment by byronyi** [comment]
Per b7796f3c856965f68699b2b527fc2872e2aa71ad perhaps you'd like to update https://github.com/tensorflow/tensorflow/blob/a83686471143b3f3234f23637c85c6b0cab3c3ef/third_party/gpus/find_cuda_config.py.gz.base64 (possibly with help from @buchgr).

Also I would suggest using the `realpath` function from https://github.com/tensorflow/tensorflow/blob/a83686471143b3f3234f23637c85c6b0cab3c3ef/third_party/remote_config/common.bzl#L254 in `third_party/gpus/cuda_configure.bzl`, which plays nice with remote builds.

**Comment by byronyi** [comment]
In `find_cuda_config.py` it seems better to change https://github.com/tensorflow/tensorflow/blob/a83686471143b3f3234f23637c85c6b0cab3c3ef/third_party/gpus/find_cuda_config.py#L438-L443 from `normpath` to `realpath`.

**Comment by njzjz** [comment]
> Per [b7796f3](https://github.com/tensorflow/tensorflow/commit/b7796f3c856965f68699b2b527fc2872e2aa71ad) perhaps you'd like to update https://github.com/tensorflow/tensorflow/blob/a83686471143b3f3234f23637c85c6b0cab3c3ef/third_party/gpus/find_cuda_config.py.gz.base64 (possibly with help from @buchgr).
> 
> Also I would suggest using the `realpath` function from
> 
> https://github.com/tensorflow/tensorflow/blob/a83686471143b3f3234f23637c85c6b0cab3c3ef/third_party/remote_config/common.bzl#L254
> in `third_party/gpus/cuda_configure.bzl`, which plays nice with remote builds.

These suggestions have been applied. I also updated `find_cuda_config.py.gz.base64` using `compress_find_cuda_config.py`.

**Comment by buchgr** [comment]
LGTM. Thanks @njzjz. Apologies for the gzip compression step. This should go away soon!

File: third_party/gpus/cuda_configure.bzl
Comment (byronyi) [inline third_party/gpus/cuda_configure.bzl]: This should be sorted in alphabetical order for linter and code-style check.

Comment (njzjz) [reply third_party/gpus/cuda_configure.bzl]: Resolved.

**Comment by angerson** [comment]
Looks good to me, but I'll defer to @byronyi for the review. Thanks for your work on this!

**Comment by byronyi** [comment]
> Looks good to me, but I'll defer to @byronyi for the review. Thanks for your work on this!

@angerson LGTM, but I am not a committer to approve this :) Mind to approve?

**Review by byronyi — DISMISSED** [review]

**Comment by njzjz** [comment]
Looks like the `realpath` function does not work if the path contains spaces.
https://github.com/tensorflow/tensorflow/blob/a83686471143b3f3234f23637c85c6b0cab3c3ef/third_party/remote_config/common.bzl#L268

I think we should add double quotes
```py
 return execute(repository_ctx, [bash_bin, "-c", "realpath \"%s\"" % path]).stdout.strip() 
```

**Comment by byronyi** [comment]
> Looks like the `realpath` function does not work if the path contains spaces.
> 
> https://github.com/tensorflow/tensorflow/blob/a83686471143b3f3234f23637c85c6b0cab3c3ef/third_party/remote_config/common.bzl#L268
> 
> I think we should add double quotes
> 
> ```python
>  return execute(repository_ctx, [bash_bin, "-c", "realpath \"%s\"" % path]).stdout.strip() 
> ```

Mind to fix this? I believe it’s the reason why CI fails on Windows.

**Review by byronyi — APPROVED** [review]

**Comment by byronyi** [comment]
Ping @buchgr; mind to kick off CI?

**Comment by byronyi** [comment]
On Windows it fails, but it does not seem related to this PR anymore.

Ping @gbaned to approve.

**Comment by buchgr** [comment]
I am not part of the Tensorflow team. I believe that I can't trigger a build.

**Comment by njzjz** [comment]
What is `import/copybara — An error happened while migrating the change`? Does anyone know?

**Comment by byronyi** [comment]
@njzjz it means it’s processing internally inside Google and we just need to be patient as external contributors.

**Review by angerson — APPROVED** [review]

**Pull request merged**
